### PR TITLE
2021-06023-Haven-Protocol

### PR DESCRIPTION
**Pull request by lz-tmz**
added article in wiki

### CODE REVIEW

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+-**June 25, 2021 (Block 884689, 07:04:19):**
+    Similar to the previous exploit, the attacker manipulated transaction types, resulting in the loss of 110 xBTC.
+
+-**June 29, 2021 (Block 887361, 00:45:20 - Block 887409, 02:15:23):**
+    The attacker exploited a vulnerability to manipulate the output values and mint arbitrary amounts. This exploit occurred 18 times within this timeframe.
+
+## Security Failure Causes
+
+1. Improve Development Processes:
+   - Open up the repository to more developers and ensure git history is included. See [GitHub](https://github.com/).
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:45]: The text is unrelated to the incident, please, rewrite the whole article. Take a look into approved articles, to find out what is wrong with the structure too.

**Review by JediFaust — CHANGES_REQUESTED** [review]

**Review by jhirschkorn — CHANGES_REQUESTED** [review]
Fix timeline bullet format

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## [Losses](https://cdn.havenprotocol.org/app/uploads/2021/07/Technical-Overview-of-June-2021-Exploits.pdf)
+
+- Approximately 125.66 xBTC with an estimated value of around $4,335,000 was obtained through the exploits.
+- An amount of 202,920 xUSD, equivalent to approximately $202,920.
+- A total of 440,000 XHV with an approximate value of $1,540,000 was frozen in KuCoin.
+- TradeOgre froze 100,000 XHV, valued at approximately $350,000.
+
+## Timeline
+
+-**June 22, 2021  18:19:41:**
```
Comment (jhirschkorn) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:24]: If including the actual time is necessary, also include the relevant time zone for clarity.

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+The identity of the attackers remains unknown.
+
+## [Losses](https://cdn.havenprotocol.org/app/uploads/2021/07/Technical-Overview-of-June-2021-Exploits.pdf)
+
+- Approximately 125.66 xBTC with an estimated value of around $4,335,000 was obtained through the exploits.
+- An amount of 202,920 xUSD, equivalent to approximately $202,920.
+- A total of 440,000 XHV with an approximate value of $1,540,000 was frozen in KuCoin.
+- TradeOgre froze 100,000 XHV, valued at approximately $350,000.
+
+## Timeline
```
Comment (jhirschkorn) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:22]: Clean up the timeline. Too verbose. Make it shorter.

**Review by jhirschkorn — CHANGES_REQUESTED** [review]

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Summary
+
+Haven Protocol experienced a series of exploits that resulted in financial losses and raised concerns about the security of the platform. The exploits targeted various aspects of the protocol, including miner rewards, transaction validation, and conversion fees.
+
+## Attackers
+
+The identity of the attackers remains unknown.
+
+## Losses
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:15]: Add total lost amount

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+---
+date: 2021-06-23
+categories: Protocol Exploit
+title: "Haven Protocol Hack"
+---
+
+## Summary
+
+Haven Protocol experienced a series of exploits that resulted in financial losses and raised concerns about the security of the platform. The exploits targeted various aspects of the protocol, including miner rewards, transaction validation, and conversion fees.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:9]: Add more technical details with reliable sources

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+title: "Haven Protocol Hack"
+---
+
+## Summary
+
+Haven Protocol experienced a series of exploits that resulted in financial losses and raised concerns about the security of the platform. The exploits targeted various aspects of the protocol, including miner rewards, transaction validation, and conversion fees.
+
+## Attackers
+
+The identity of the attackers remains unknown.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:13]: Add attacker's address

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+- TradeOgre froze 100,000 XHV, valued at approximately $350,000.
+
+## Timeline
+
+- **June 22, 2021:** Miner exploited a vulnerability, minting higher rewards (6.73 xBTC, 101,460 xUSD).
+- **June 23, 2021:** Similar exploit, increased block rewards obtained (6.73 xBTC, 101,460 xUSD).
+- **June 24, 2021:** Attacker modified outputs, minted extra xAssets (2.2 xBTC). Invalid transaction led to loss of counterfeit funds.
+- **June 25, 2021:** Attacker manipulated transaction types, resulting in loss of 110 xBTC.
+- **June 29, 2021:** Exploitation of vulnerability, minting arbitrary amounts occurred 18 times.
+- **July 19, 2021:** Chain rollback through a [fork](https://havenprotocol.medium.com/haven-protocol-successfully-deploys-rollback-hard-fork-206e5ead190e), allowing exchange wallets to reopen and on-chain transactions to resume. xUSD and xAsset conversions in the Haven Vault remain paused.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:29]: Shorten timeline, providing important events and sources. Look up on other approved articles as a reference

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+## Security Failure Causes
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:31]: Add sources and make bullet point as in other approved articles

**Review by JediFaust — CHANGES_REQUESTED** [review]

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 9]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
 
 ## Summary
 
-Haven Protocol experienced a series of exploits that resulted in financial losses and raised concerns about the security of the platform. The exploits targeted various aspects of the protocol, including miner rewards, transaction validation, and conversion fees.
+Haven Protocol experienced a series of exploits that resulted in financial losses. The attacker [manipulated](https://havenprotocol.org/wp-content/uploads/2021/07/Technical-Overview-of-June-2021-Exploits.pdf) their transaction by modifying the proof-of-value and burnt/mint checks, setting the amount burnt/mint to 0. Since multiplying 0 by any value results in 0, the validation passed, allowing the attacker to proceed. By manipulating the output values, the attacker was able to mint an arbitrary amount.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:9]: There is no such thing as 'manipulating transaction'. Also, source is not accessible by now.

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 10]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
 
 ## Summary
 
-Haven Protocol experienced a series of exploits that resulted in financial losses and raised concerns about the security of the platform. The exploits targeted various aspects of the protocol, including miner rewards, transaction validation, and conversion fees.
+Haven Protocol experienced a series of exploits that resulted in financial losses. The attacker [manipulated](https://havenprotocol.org/wp-content/uploads/2021/07/Technical-Overview-of-June-2021-Exploits.pdf) their transaction by modifying the proof-of-value and burnt/mint checks, setting the amount burnt/mint to 0. Since multiplying 0 by any value results in 0, the validation passed, allowing the attacker to proceed. By manipulating the output values, the attacker was able to mint an arbitrary amount.
 
 ## Attackers
 
-The identity of the attackers remains unknown.
+The identity of the attackers remains unknown. The attacker address involved in the Haven Protocol exploit is not publicly disclosed or known.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:13]: Remove the last sentence, was thinking the address is known.

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 11]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
...
-The identity of the attackers remains unknown.
+The identity of the attackers remains unknown. The attacker address involved in the Haven Protocol exploit is not publicly disclosed or known.
 
 ## Losses
 
+- The total losses amount to approximately $6,427,920
 - Approximately 125.66 xBTC with an estimated value of around $4,335,000 was obtained through the [exploits](https://cdn.havenprotocol.org/app/uploads/2021/07/Technical-Overview-of-June-2021-Exploits.pdf).
 - An amount of 202,920 xUSD, equivalent to approximately $202,920.
 - A total of 440,000 XHV with an approximate value of $1,540,000 was frozen in KuCoin.
 - TradeOgre froze 100,000 XHV, valued at approximately $350,000.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:21]: Reformat as `$4,335,000 worth 125.66 xBTC` for all further lines

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 12]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
...
 
 ## Timeline
 
-- **June 22, 2021:** Miner exploited a vulnerability, minting higher rewards (6.73 xBTC, 101,460 xUSD).
-- **June 23, 2021:** Similar exploit, increased block rewards obtained (6.73 xBTC, 101,460 xUSD).
-- **June 24, 2021:** Attacker modified outputs, minted extra xAssets (2.2 xBTC). Invalid transaction led to loss of counterfeit funds.
-- **June 25, 2021:** Attacker manipulated transaction types, resulting in loss of 110 xBTC.
-- **June 29, 2021:** Exploitation of vulnerability, minting arbitrary amounts occurred 18 times.
+- **June 22, 2021:** 203,000 xUSD and 13.5 xBTC was minted in two [exploits](https://havenprotocol.medium.com/haven-protocol-exploits-mitigation-plan-and-next-steps-d7a2b1a65654). 
+- **June 24, 2021:** An exploit in the xAsset conversion validation.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:26]: Rewrite the line with your own words

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 13]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
...
+- **June 29, 2021:** an exploit was leveraged that allowed for minting of 9m xUSD.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:27]: Rewrite with your own words

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 14]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
...
 - **July 19, 2021:** Chain rollback through a [fork](https://havenprotocol.medium.com/haven-protocol-successfully-deploys-rollback-hard-fork-206e5ead190e), allowing exchange wallets to reopen and on-chain transactions to resume. xUSD and xAsset conversions in the Haven Vault remain paused.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:28]: Rewrite like 'Haven Protocol performed a rollback'

File: content/attacks/posts/2021-06-23-Haven-Protocol.md [hunk 15]
```diff
@@ -6,30 +6,30 @@ title: "Haven Protocol Hack"
...
 
 ## Security Failure Causes
 
-- Inadequate Development Processes: Insufficient attention to development practices and protocols, leading to vulnerabilities and exploits. Lack of open repository, limited unit testing, and incomplete code reviews contributed to the security breach.
-
-- Lack of Community Engagement: Insufficient community involvement and collaboration hindered the identification of potential vulnerabilities. The absence of bug bounty programs, infrequent technical calls, and a lack of transparent proposal discussions limited the collective effort to enhance security.
+- Insufficient attention to development practices.
```
Comment (JediFaust) [inline content/attacks/posts/2021-06-23-Haven-Protocol.md:32]: Format bullet points and return explanations. Just look up on approved articles in the main branch.

**Review by JediFaust — CHANGES_REQUESTED** [review]

**Comment by JediFaust** [comment]
No update for 2 weeks. Closing.

**Pull request closed without merging**